### PR TITLE
Fix node syncing

### PR DESCRIPTION
**Pull request by ales-tsurko**
Currently our nodes panic ([1] [2]) when we try to sync them from scratch. The issue is happened because of multiple issues:

- we introduced a runtime upgrade without incrementing the chain spec version (#99);
- we patched `pallet_democracy` and because our nodes couldn't produce blocks without the patch, we used [`codeSubstitutes`](https://paritytech.github.io/polkadot-sdk/master/sc_service/trait.ChainSpec.html#tymethod.code_substitutes) in the chainspec.

All those upgrades happened at the same spec version and without spec version update. And it turned out, it's only possible to have [a single substitute per spec version](https://github.com/paritytech/polkadot-sdk/blob/release-polkadot-v1.11.0/substrate/client/service/src/client/wasm_substitutes.rs#L96), thus using another substitute for the first upgrade provides random substitute usage (either the first or the second one), but not both.

This PR provides:

- switching to `WasmExecutor` as suggested in [2] as `NativeElseWasm` may lead to [similar issues](https://github.com/paritytech/polkadot-sdk/issues/4808#issuecomment-2181237429);
- switching to our fork of `polkadot-sdk`, where `WasmSubstitutes` can hold multiple substitutes per the same spec version;
- switching to our fork of `frontier`, because it depends on `polkadot-sdk` repo, but we need it to use our fork (it's easier to update **Cargo.toml** in the fork, than find out which dependencies we need to **patch**);
- adding [**bad blocks**](https://paritytech.github.io/polkadot-sdk/master/sc_client_api/client/type.BadBlocks.html) and [**fork blocks**](https://paritytech.github.io/polkadot-sdk/master/sc_client_api/client/type.ForkBlocks.html) chainspec extensions;
- updating the chainspec file for testnet with code substitutes;
- removing `pallet-democracy-patched` as the fix is now in the forked substrate.

Intention behind the fork is caused by the lack of related substrate API, which is neither parametrized nor public. Particularly `WasmSubstitute` and parts of `WasmSubstitutes` which holds a map with the spec version as a key and the substitute as a value.

[1]: https://substrate.stackexchange.com/questions/11674/cant-sync-with-network
[2]: https://github.com/paritytech/polkadot-sdk/issues/5008

### CODE REVIEW

File: scripts/run_testnet.sh [hunk 1]
```diff
@@ -52,39 +53,26 @@ stop_network() {
...
         --rpc-cors=all \
         --validator \
-        --rpc-port $rpc_port \
+        --state-pruning archive \
+        --allow-private-ipv4 \
+        --rpc-port "$rpc_port" \
         --base-path "${base_path}/node-${rpc_port}/" \
-        --port $p2p_port \
-        --bootnodes /ip4/127.0.0.1/tcp/30333/p2p/12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp &
+        --unsafe-force-node-key-generation \
```
Comment (DmitriVT) [inline scripts/run_testnet.sh:74]: Automatically generate node keys: this is not recommended for testnet/mainnet to security risks.
Can we avoid using --unsafe-force-node-key-generation by securely generating keys manually?

For example using a script:
`
/node-data key generate-node-key --file /noda-data/node-key
`

File: node/src/service.rs [hunk 2]
```diff
@@ -194,27 +193,22 @@ where
...
-    RuntimeApi: ConstructRuntimeApi<Block, FullClient<RuntimeApi, Executor>>,
-    RuntimeApi: Send + Sync + 'static,
-    RuntimeApi::RuntimeApi: RuntimeApiCollection,
-    Executor: NativeExecutionDispatch + 'static,
-{
-    let _frontier_block_import =
-        FrontierBlockImport::new(grandpa_block_import.clone(), client.clone());
+> {
+    // TODO should we use this instead of babe block import?
+    // let _frontier_block_import =
```
Comment (DmitriVT) [inline node/src/service.rs:210]: This code creates a new instance of the Frontier block importer, which is used to support Ethereum-compatible functionality. The importer integrates with the existing GRANDPA consensus mechanism using grandpa_block_import and client. 
If we remove this import, are we sure that Ethereum transactions and smart contracts will be supported?

**Review by DmitriVT — CHANGES_REQUESTED** [review]

File: node/src/service.rs (hunk 2 above)
Comment (ales-tsurko) [reply node/src/service.rs:210]: Please, check further. It has been abandoned for a while. It’s not used in the code - babe import queue is used.

Comment (ales-tsurko) [reply node/src/service.rs:210]: The code wouldn’t compile if I randomly disabled variable which used somewhere later

**Review by DmitriVT — APPROVED** [review]

**Review by hrls — APPROVED** [review]

**Pull request merged**